core/modules/wdb/ChunkResource.cc

```diff
@@ -235,8 +279,112 @@ class Backend {
+        int uid = atoi(uidStr.c_str());
+        if (uid != _uid) {
+            std::ostringstream emsg;
+            emsg << "memLockStatus LOCKED_OTHER wrong uid. Expected " << _uid << " got " << uid
+                << " err=" << err.printErrMsg();
+            LOGF_WARN("%1%" % emsg.str());
+            return LOCKED_OTHER;
+        }
+        return LOCKED_OURS;
+    }
+
+    /// Take the lock on the in memory tables, even if held by another.
+    // Terminate the program if we can't take the lock.
+    // This must be run before any other operations on in memory tables.
+    void _memLockLock() {
+        _lockDb = MEMLOCKDB;
+        _lockTbl = MEMLOCKTBL;
+        _lockDbTbl = _lockDb + "." + _lockTbl;
+        LockStatus mls = _memLockStatus();
+        if (mls != UNLOCKED) {
+            LOGF_WARN("Memory tables were not released cleanly! LockStatus=%1%" % strLockStatus(mls));
+        }
+
+        // Lock the memory tables.
+        std::string sql = "CREATE DATABASE IF NOT EXISTS " + _lockDb + ";";
+        sql += "CREATE TABLE IF NOT EXISTS " + _lockDbTbl + " ( keyId INT UNIQUE, uid INT ) ENGINE = MEMORY;";
+        _execLockSql(sql);
+        sql = "TRUNCATE TABLE " + _lockDbTbl;
+        _execLockSql(sql);
+        std::ostringstream insert;
+        insert << "INSERT INTO " << _lockDbTbl << " (keyId, uid) VALUES(1, " << _uid << " );";
+        _execLockSql(insert.str());
+
+        // Delete any old in memory databases. They could be empty or wrong.
+        // Empty tables would prevent new tables from being created.
+        std::ostringstream os;
+        os << "SHOW DATABASES LIKE '" << SUBCHUNKDB_PREFIX << "%';";
+        sql::SqlResults results;
+        sql::SqlErrorObject err;
+        if (!_sqlConn.runQuery(os.str(), results, err)) {
+            std::ostringstream emsg;
+            emsg << "Memory database query failed, exiting. " << os.str() << " err=" << err.printErrMsg();
+            LOGF_ERROR("%1%" % emsg.str());
+            exit(EXIT_FAILURE);
+        }
+        std::vector<std::string> databases;
+        results.extractFirstColumn(databases, err);
+        for (auto iter=databases.begin(), end=databases.end(); iter != end;) {
+            // Delete in blocks of 50 to save time.
+            std::string dropDb = "";
+            int count = 0;
+            while (iter != end && count < 50) {
+                std::string db = *iter;
+                dropDb += "DROP DATABASE " + db + ";";
+                ++iter;
+                ++count;
+            }
+            _execLockSql(dropDb);
+        }
+    }
+
+    /// Delete the memory lock database and everything in it.
+    void _memLockRelease() {
+        LOGF_DEBUG("memLockRelease");
+        std::string sql = "DROP DATABASE " + _lockDb + ";";
+        _execLockSql(sql);
+    }
+
     bool _isFake;
     sql::SqlConnection _sqlConn;
+
+    // Memory lock table members.
+    std::string _lockDb;
+    std::string _lockTbl;
+    std::string _lockDbTbl;
```

Comment:
Can't you get rid of all the `_lockXXX` member fields and just use the constants you've defined elsewhere?